TimefoldProcessor#recordAndRegisterDevUIBean NullPointerException in fresh Quarkus project

**Describe the bug**
I'm creating a Timefold project from scratch (not using the quickstarts repo) and got `java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step ai.timefold.solver.quarkus.deployment.TimefoldProcessor#recordAndRegisterDevUIBean threw an exception: java.lang.NullPointerException: Cannot read field "generatedGizmoMemberAccessorClassSet" because the return value of "ai.timefold.solver.quarkus.deployment.SolverConfigBuildItem.getGeneratedGizmoClasses()" is null`


**To Reproduce**

1. Create a fresh Quarkus project (code.quarkus.io) with the following extensions selected:

- Timefold benchmark [timefold-solver-quarkus-benchmark:1.18.0]
- Timefold Jackson [timefold-solver-quarkus-jackson:1.18.0]
- Hibernate Validator [quarkus-hibernate-validator]
- REST [quarkus-rest]
- REST Jackson [quarkus-rest-jackson]
- Web Bundler [quarkus-web-bundler:1.8.1]
- YAML Configuration [quarkus-config-yaml]

2. Start Quarkus application

```
$ cd quarkusproject
$ quarkus dev
```


## Environment

**Quarkus application.yaml**

```yml
greeting:
  message: "hello"
quarkus:
  web-bundler:
    dependencies:
      node-modules: "node_modules"
  timefold:
    solver:
      domain-access-type: REFLECTION
```

I added `quarkus.timefold.solver.domain-access-type` because I thought it would fix the issue

**Timefold Solver Version or Git ref**:

```
ai.timefold.solver:timefold-solver-quarkus:1.18.0        
ai.timefold.solver:timefold-solver-quarkus-benchmark:1.18.0
ai.timefold.solver:timefold-solver-quarkus-jackson:1.18.0
```

**Output of `java -version`:**
```
openjdk 21.0.1 2023-10-17 LTS
OpenJDK Runtime Environment Zulu21.30+15-CA (build 21.0.1+12-LTS)
OpenJDK 64-Bit Server VM Zulu21.30+15-CA (build 21.0.1+12-LTS, mixed mode, sharing)
```

**Output of `uname -a` or `ver`:**
```
Darwin Skynet 24.2.0 Darwin Kernel Version 24.2.0: Fri Dec  6 19:01:59 PST 2024; root:xnu-11215.61.5~2/RELEASE_ARM64_T6000 arm64
```

## Additional information

Provide any and all other information which might be relevant to the issue.

**Quarkus Version:**
```
3.16.1
```

**Maven version**
```
Apache Maven 3.9.9 (8e8579a9e76f7d015ee5ec7bfcdc97d260186937)
Maven home: /Users/ad/.m2/wrapper/dists/apache-maven-3.9.9-bin/33b4b2b4/apache-maven-3.9.9
Java version: 21.0.1, vendor: Azul Systems, Inc., runtime: /Users/ad/.sdkman/candidates/java/21.0.1-zulu/zulu-21.jdk/Contents/Home
Default locale: en_MY, platform encoding: UTF-8
OS name: "mac os x", version: "15.2", arch: "aarch64", family: "mac"
```

**Full pom.xml**
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>com.xxxxxx</groupId>
    <artifactId>quarkusnew</artifactId>
    <version>1.0.0-SNAPSHOT</version>

    <properties>
        <compiler-plugin.version>3.13.0</compiler-plugin.version>
        <maven.compiler.release>21</maven.compiler.release>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
        <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
        <quarkus.platform.group-id>io.quarkus.platform</quarkus.platform.group-id>
        <quarkus.platform.version>3.17.8</quarkus.platform.version>
        <skipITs>true</skipITs>
        <surefire-plugin.version>3.5.0</surefire-plugin.version>
    </properties>

    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>${quarkus.platform.group-id}</groupId>
                <artifactId>${quarkus.platform.artifact-id}</artifactId>
                <version>${quarkus.platform.version}</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
        </dependencies>
    </dependencyManagement>

    <dependencies>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-rest</artifactId>
        </dependency>
        <dependency>
            <groupId>ai.timefold.solver</groupId>
            <artifactId>timefold-solver-quarkus</artifactId>
            <version>1.18.0</version>
        </dependency>
        <dependency>
            <groupId>ai.timefold.solver</groupId>
            <artifactId>timefold-solver-quarkus-benchmark</artifactId>
            <version>1.18.0</version>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-hibernate-validator</artifactId>
        </dependency>
        <dependency>
            <groupId>ai.timefold.solver</groupId>
            <artifactId>timefold-solver-quarkus-jackson</artifactId>
            <version>1.18.0</version>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-rest-jackson</artifactId>
        </dependency>
        <dependency>
            <groupId>io.quarkiverse.web-bundler</groupId>
            <artifactId>quarkus-web-bundler</artifactId>
            <version>1.8.1</version>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-config-yaml</artifactId>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-arc</artifactId>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-junit5</artifactId>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>io.rest-assured</groupId>
            <artifactId>rest-assured</artifactId>
            <scope>test</scope>
        </dependency>
    </dependencies>

    <build>
        <plugins>
            <plugin>
                <groupId>${quarkus.platform.group-id}</groupId>
                <artifactId>quarkus-maven-plugin</artifactId>
                <version>${quarkus.platform.version}</version>
                <extensions>true</extensions>
                <executions>
                    <execution>
                        <goals>
                            <goal>build</goal>
                            <goal>generate-code</goal>
                            <goal>generate-code-tests</goal>
                            <goal>native-image-agent</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
            <plugin>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>${compiler-plugin.version}</version>
                <configuration>
                    <parameters>true</parameters>
                </configuration>
            </plugin>
            <plugin>
                <artifactId>maven-surefire-plugin</artifactId>
                <version>${surefire-plugin.version}</version>
                <configuration>
                    <systemPropertyVariables>
                        <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
                        <maven.home>${maven.home}</maven.home>
                    </systemPropertyVariables>
                </configuration>
            </plugin>
            <plugin>
                <artifactId>maven-failsafe-plugin</artifactId>
                <version>${surefire-plugin.version}</version>
                <executions>
                    <execution>
                        <goals>
                            <goal>integration-test</goal>
                            <goal>verify</goal>
                        </goals>
                    </execution>
                </executions>
                <configuration>
                    <systemPropertyVariables>
                        <native.image.path>${project.build.directory}/${project.build.finalName}-runner</native.image.path>
                        <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
                        <maven.home>${maven.home}</maven.home>
                    </systemPropertyVariables>
                </configuration>
            </plugin>
        </plugins>
    </build>

    <profiles>
        <profile>
            <id>native</id>
            <activation>
                <property>
                    <name>native</name>
                </property>
            </activation>
            <properties>
                <skipITs>false</skipITs>
                <quarkus.native.enabled>true</quarkus.native.enabled>
            </properties>
        </profile>
    </profiles>
</project>
```

**Full stack trace**

```
[INFO] Scanning for projects...
[INFO]
[INFO] ---------------------< com.xxxxx:quarkusnew >---------------------
[INFO] Building quarkusnew 1.0.0-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- resources:3.3.1:resources (default-resources) @ quarkusnew ---
[INFO] Copying 5 resources from src/main/resources to target/classes
[INFO]
[INFO] --- quarkus:3.17.8:generate-code (default) @ quarkusnew ---
[INFO]
[INFO] --- compiler:3.13.0:compile (default-compile) @ quarkusnew ---
[INFO] Nothing to compile - all classes are up to date.
[INFO]
[INFO] --- quarkus:3.17.8:dev (default-cli) @ quarkusnew ---
[INFO] Invoking resources:3.3.1:testResources (default-testResources) @ quarkusnew
[INFO] skip non existing resourceDirectory /Users/ad/Development/quarkusnew/src/test/resources
[INFO] Invoking quarkus:3.17.8:generate-code-tests (default) @ quarkusnew
[INFO] Invoking compiler:3.13.0:testCompile (default-testCompile) @ quarkusnew
[INFO] Nothing to compile - all classes are up to date.
[WARNING] Changed debug port to 49762 because of a port conflict
Listening for transport dt_socket at address: 49762

2025-01-29 17:57:04,249 WARN  [ai.tim.sol.qua.dep.TimefoldProcessor] (build-10) Skipping Timefold extension because there are no @PlanningSolution or @PlanningEntity annotated classes.
If your domain classes are located in a dependency of this project, maybe try generating the Jandex index by using the jandex-maven-plugin in that dependency, or by addingapplication.properties entries (quarkus.index-dependency.<name>.group-id and quarkus.index-dependency.<name>.artifact-id).

2025-01-29 17:57:04,250 WARN  [ai.tim.sol.ben.qua.dep.TimefoldBenchmarkProcessor] (build-42) Skipping Timefold Benchmark extension because the Timefold extension was skipped.
2025-01-29 17:57:04,267 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start live reload endpoint to recover from previous Quarkus startup failure
2025-01-29 17:57:04,506 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step ai.timefold.solver.quarkus.deployment.TimefoldProcessor#recordAndRegisterDevUIBean threw an exception: java.lang.NullPointerException: Cannot read field "generatedGizmoMemberAccessorClassSet" because the return value of "ai.timefold.solver.quarkus.deployment.SolverConfigBuildItem.getGeneratedGizmoClasses()" is null
	at ai.timefold.solver.quarkus.deployment.TimefoldProcessor.recordAndRegisterDevUIBean(TimefoldProcessor.java:651)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
	at java.base/java.lang.Thread.run(Thread.java:1583)
	at org.jboss.threads.JBossThread.run(JBossThread.java:499)

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:355)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:272)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:62)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:89)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:428)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:55)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:138)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:93)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:131)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step ai.timefold.solver.quarkus.deployment.TimefoldProcessor#recordAndRegisterDevUIBean threw an exception: java.lang.NullPointerException: Cannot read field "generatedGizmoMemberAccessorClassSet" because the return value of "ai.timefold.solver.quarkus.deployment.SolverConfigBuildItem.getGeneratedGizmoClasses()" is null
	at ai.timefold.solver.quarkus.deployment.TimefoldProcessor.recordAndRegisterDevUIBean(TimefoldProcessor.java:651)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
	at java.base/java.lang.Thread.run(Thread.java:1583)
	at org.jboss.threads.JBossThread.run(JBossThread.java:499)

	at io.quarkus.builder.Execution.run(Execution.java:124)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:161)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:351)
	... 9 more
Caused by: java.lang.NullPointerException: Cannot read field "generatedGizmoMemberAccessorClassSet" because the return value of "ai.timefold.solver.quarkus.deployment.SolverConfigBuildItem.getGeneratedGizmoClasses()" is null
	at ai.timefold.solver.quarkus.deployment.TimefoldProcessor.recordAndRegisterDevUIBean(TimefoldProcessor.java:651)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
	at java.base/java.lang.Thread.run(Thread.java:1583)
	at org.jboss.threads.JBossThread.run(JBossThread.java:499)
```

--- a/quarkus-integration/quarkus/deployment/src/main/java/ai/timefold/solver/quarkus/deployment/TimefoldProcessor.java
+++ b/quarkus-integration/quarkus/deployment/src/main/java/ai/timefold/solver/quarkus/deployment/TimefoldProcessor.java
@@ -8,6 +8,7 @@ import java.lang.reflect.Method;
 import java.lang.reflect.Modifier;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.LinkedHashSet;
@@ -669,6 +670,18 @@ class TimefoldProcessor {
             RecorderContext recorderContext,
             SolverConfigBuildItem solverConfigBuildItem,
             BuildProducer<SyntheticBeanBuildItem> syntheticBeans) {
+        if (solverConfigBuildItem.getGeneratedGizmoClasses() == null) {
+            // Extension was skipped, so no solver configs
+            syntheticBeans.produce(SyntheticBeanBuildItem.configure(DevUISolverConfig.class)
+                    .scope(ApplicationScoped.class)
+                    .supplier(devUIRecorder.solverConfigSupplier(Collections.emptyMap(),
+                            Collections.emptyMap(),
+                            Collections.emptyMap()))
+                    .defaultBean()
+                    .setRuntimeInit()
+                    .done());
+            return;
+        }
         syntheticBeans.produce(SyntheticBeanBuildItem.configure(DevUISolverConfig.class)
                 .scope(ApplicationScoped.class)
                 .supplier(devUIRecorder.solverConfigSupplier(solverConfigBuildItem.getSolverConfigMap(),

--- a/quarkus-integration/quarkus/deployment/src/test/java/ai/timefold/solver/quarkus/TimefoldProcessorDevModeEmptyAppTest.java
+++ b/quarkus-integration/quarkus/deployment/src/test/java/ai/timefold/solver/quarkus/TimefoldProcessorDevModeEmptyAppTest.java
@@ -1,0 +1,22 @@
+package ai.timefold.solver.quarkus;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusDevModeTest;
+
+public class TimefoldProcessorDevModeEmptyAppTest {
+
+    @RegisterExtension
+    static final QuarkusDevModeTest config = new QuarkusDevModeTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses());
+
+    @Test
+    void emptyAppDoesNotCrash() {
+        // Success if it didn't crash during bootstrap
+    }
+
+}

